internals-js/src/coreSpec.ts

```diff
@@ -288,43 +297,61 @@ export function isCoreSpecDirectiveApplication(directive: Directive<SchemaDefini
   }
 }
 
+const linkPurposeTypeSpec = createEnumTypeSpecification({
+  name: 'Purpose',
+  values: corePurposes.map((name) => ({ name, description: purposesDescription(name)}))
+});
+
+const linkImportTypeSpec = createScalarTypeSpecification({ name: 'Import' });
+
 export class CoreSpecDefinition extends FeatureDefinition {
+  private readonly directiveDefinitionSpec: DirectiveSpecification;
+
   constructor(version: FeatureVersion, identity: string = linkIdentity, name: string = linkDirectiveDefaultName) {
     super(new FeatureUrl(identity, name, version));
+    this.directiveDefinitionSpec = createDirectiveSpecification({
+      name,
+      locations: [DirectiveLocation.SCHEMA],
+      repeatable: true,
+      argumentFct: (schema, nameInSchema) => this.createDefinitionArgumentSpecifications(schema, nameInSchema),
+    });
+  }
+
+  private createDefinitionArgumentSpecifications(schema: Schema, nameInSchema?: string): { args: ArgumentSpecification[], errors: GraphQLError[] } {
+    const args: ArgumentSpecification[] = [
+      { name: this.urlArgName(), type: new NonNullType(schema.stringType()) },
```

Comment:
can we weaken this to just `String` for the moment? i'd like v1.0 to maintain universality. there are a few mechanisms to do this, but `@link(url: null, import: "looks__namespacedButIsnt")` is one way to handle it, and i'd like us not to have painted ourselves into a corner by shipping it like this.

Reply:
Make sense. And was simple-enough, so made the change in the PR (with a test to double-check it's a safe change, that is that we were still correctly reading previously generated definition having the non-nullable type).